contributors/design-proposals/namespace-specific-dns.md

```diff
@@ -0,0 +1,55 @@
+#Description
+
+Multi-tenant is a software architecture which describes that one single instance of software serves multiple customers at the same time. This architecture provides isolated view of the whole system to each of the customer so that they co-exist in the same system and won’t affect, or see, each other. 
+
+Kubernetes as a container orchestrator platform, uses namespace to realize multi-tenant architecture. But for now, there is only one cluster level DNS server deployed in the kubernetes environment, which means that pods can find/resolve-to the service that is from another namespace.
+
+To enhance the multi-tenant support, this proposal suggests to support namespace specific DNS servers. The use case for this proposal is a kubernetes operator who is providing a multi-tenant deployment environment based on namespaces. Each namespace belongs to one tenant. The operator can deploy separate DNS servers for different namespaces, so each tenant gets its own namespace and DNS server. In this case, users from other namespaces won't find/resolve-to the services defined in our own namespace. This will help keep the privacy of the multi-tenant environment.
```

Comment:
@thockin just to make sure I understand your suggestion, rather than having the DNS server be provided the name of a namespace, you're suggesting we have a label-selector (over namespaces) provided instead, right?